src/celeritas/optical/model/RayleighModel.cc

```diff
@@ -111,6 +105,27 @@ void RayleighModel::build_mfps(OptMatId mat, MfpBuilder& build) const
         }
         build(grid);
     }
+    // Build a grid with infinite MFP to prevent selection of the model
+    else
+    {
+        auto mat_view = input_.materials->get(mat);
+        auto rindex_calc = mat_view.make_refractive_index_calculator();
+        auto energies = rindex_calc.grid().values();
+
+        if (!energies.empty())
+        {
+            inp::Grid g;
+
+            // Using min and max energy values from the refractive index grid
+            // and setting MFP to infinity
+            g.x = {energies.front(), energies.back()};
+            g.y = {std::numeric_limits<real_type>::infinity(),
+                   std::numeric_limits<real_type>::infinity()};
+            CELER_LOG(debug) << "Material " << mat.get()
+                             << " has no MFP data - setting mfp to infinity";
```

Comment:
Minor change for consistency with our usual messages
```suggestion
                             << " has no MFP data: setting mfp to infinity";
```